WalletWasabi.Fluent/Models/Wallets/TransactionTreeBuilder.cs

```diff
@@ -150,7 +150,7 @@ private TransactionModel CreateRegular(int index, TransactionSummary transaction
 			Confirmations = confirmations,
 			BlockHeight = transactionSummary.Height.Type == HeightType.Chain ? transactionSummary.Height.Value : 0,
 			BlockHash = transactionSummary.BlockHash,
-			Fee = transactionSummary.GetFee(),
+			Fee = transactionSummary.GetFee() ?? transactionSummary.FetchedFee,
```

Comment:
My idea here is that:

1. We try to calculate the fee in the old way - client side.
2. If that fails we know the transaction has a foreign input, so use the `FetchedFee` which we get from backend
3. If FetchedFee is not yet ready we assign `Money.Zero`.
4. When the fee arrives from backend we rebuild the whole history, so `FetchedFee` will contain the actual fee of the TX.